testng-core/src/main/java/org/testng/internal/MethodSorting.java

```diff
@@ -0,0 +1,85 @@
+package org.testng.internal;
+
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+import org.testng.ITestNGMethod;
+
+public enum MethodSorting implements Comparator<ITestNGMethod> {
+  METHOD_NAMES("methods") {
+    @Override
+    public int compare(ITestNGMethod o1, ITestNGMethod o2) {
+      String n1 = o1.getMethodName();
+      String n2 = o2.getMethodName();
+      return n1.compareTo(n2);
+    }
+
+    @Override
+    public String toString() {
+      return "Method_Names";
+    }
+  },
+  INSTANCES("instances") {
+    @Override
+    public int compare(ITestNGMethod o1, ITestNGMethod o2) {
+      Comparator<ITestNGMethod> comparator =
+          Comparator.comparingInt(ITestNGMethod::getPriority)
+              .thenComparing(method -> method.getRealClass().getName())
+              .thenComparing(ITestNGMethod::getMethodName)
+              .thenComparing(Object::toString)
+              .thenComparing(
+                  method ->
+                      Optional.ofNullable(method.getFactoryMethodParamsInfo())
+                          .map(it -> Arrays.toString(it.getParameters()))
+                          .orElse(""))
+              .thenComparing(this::objectEquality);
+      return comparator.compare(o1, o2);
+    }
+
+    private int objectEquality(ITestNGMethod a, ITestNGMethod b) {
```

Comment:
Could be static

Reply:
Looks like that is not supposed at JDK11 and needs atleast JDK16